algorithms/src/snark/marlin/ahp/prover/oracles.rs

```diff
@@ -51,60 +51,59 @@ impl<F: PrimeField> FirstOracles<F> {
     }
 }
 
+/// The LDE of `w`.
 #[derive(Debug, Clone)]
-pub(in crate::snark::marlin) struct SingleEntry<F: PrimeField> {
-    /// The evaluations of `Az`.
-    pub(super) z_a: LabeledPolynomialWithBasis<'static, F>,
-    /// The evaluations of `Bz`.
-    pub(super) z_b: LabeledPolynomialWithBasis<'static, F>,
-    /// The LDE of `w`.
-    pub(super) w_poly: LabeledPolynomial<F>,
-    /// The LDE of `Az`.
-    pub(super) z_a_poly: LabeledPolynomial<F>,
-    /// The LDE of `Bz`.
-    pub(super) z_b_poly: LabeledPolynomial<F>,
-}
+pub(in crate::snark::marlin) struct WitnessPoly<F: PrimeField>(pub(in crate::snark::marlin) LabeledPolynomial<F>);
 
-impl<F: PrimeField> SingleEntry<F> {
+impl<F: PrimeField> WitnessPoly<F> {
     /// Iterate over the polynomials output by the prover in the first round.
     /// Intended for use when committing.
     pub fn iter_for_commit(&mut self) -> impl Iterator<Item = LabeledPolynomialWithBasis<'static, F>> {
-        let w_poly = self.w_poly.clone();
-
-        let mut z_a_copy = LabeledPolynomialWithBasis { polynomial: vec![], info: self.z_a.info().clone() };
-        std::mem::swap(&mut self.z_a, &mut z_a_copy);
+        let WitnessPoly(w_poly) = self.clone();
 
-        let mut z_b_copy = LabeledPolynomialWithBasis { polynomial: vec![], info: self.z_b.info().clone() };
-        std::mem::swap(&mut self.z_b, &mut z_b_copy);
-        [w_poly.into(), z_a_copy, z_b_copy].into_iter()
+        [w_poly.into()].into_iter()
     }
 
     /// Iterate over the polynomials output by the prover in the first round.
     /// Intended for use when opening.
     pub fn iter_for_open(&self) -> impl Iterator<Item = &LabeledPolynomial<F>> {
-        [(&self.w_poly), &self.z_a_poly, &self.z_b_poly].into_iter()
+        [(&self.0)].into_iter()
     }
```

Comment:
We can unify these functions now, since they return the same thing

Reply:
Ah, thx 8614ed484